### PR TITLE
feat(CouponList): add list-footer slot

### PR DESCRIPTION
**Pull request by Tisiven**
### Before submitting a pull request, please make sure the following is done:

1. Read the [contributing guide](https://github.com/youzan/vant/blob/dev/.github/CONTRIBUTING.md).
2. If you've added code that should be tested, add tests.
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).

#### Title Format

type(ComponentName?)：commit message

Example：

- docs: fix typo in quickstart
- build: optimize build speed
- fix(Button): incorrect style
- feat(Button): add color prop

Allowed Types:

- fix
- feat
- docs
- perf
- test
- types
- build
- chore
- refactor
- breaking change

### CODE REVIEW

File: src/coupon-list/README.zh-CN.md [hunk 1]
```diff
@@ -34,7 +34,14 @@ Vue.use(CouponList);
...
     @exchange="onExchange"
-  />
+  >
+    <template #list-footer>
+      <van-loading size="24px" class="footer">加载中...</van-loading>
+    </template>
+    <template #disabled-list-footer>
+      <div size="24px" class="footer">没有更多内容了</div>
+    </template>
+  </van-coupon-list>
```
Comment (chenjiahan) [inline src/coupon-list/README.zh-CN.md:44]: 示例部分可以删除掉，不是很通用

File: src/coupon-list/demo/index.vue [hunk 2]
```diff
@@ -20,7 +20,14 @@
...
           @exchange="onExchange"
-        />
+        >
+          <template #list-footer>
+            <van-loading size="24px" class="footer">加载中...</van-loading>
+          </template>
+          <template #disabled-list-footer>
+            <div size="24px" class="footer">没有更多内容了</div>
+          </template>
+        </van-coupon-list>
```
Comment (chenjiahan) [inline src/coupon-list/demo/index.vue:30]: 示例部分可以删除掉，不是很通用

Comment (chenjiahan) [inline src/coupon-list/demo/index.vue:30]: 英文文档里缺少对 slot 的说明

File: src/coupon-list/README.zh-CN.md (hunk 1 above)
Comment (Tisiven) [reply src/coupon-list/README.zh-CN.md:44]: 已修复

File: src/coupon-list/demo/index.vue (hunk 2 above)
Comment (Tisiven) [reply src/coupon-list/demo/index.vue]: 已修复

Comment (Tisiven) [reply src/coupon-list/demo/index.vue]: 已修复

**Pull request merged**